nixos-modules/microvm/virtiofsd/default.nix

```diff
@@ -0,0 +1,72 @@
+{ config, lib, pkgs, ... }:
+
+let
+  virtiofsShares = builtins.filter ({ proto, ... }:
+    proto == "virtiofs"
+  ) config.microvm.shares;
+
+  requiresVirtiofsd = virtiofsShares != [];
+
+  inherit (pkgs.python3Packages) supervisor;
+  supervisord = lib.getExe' supervisor "supervisord";
+  supervisorctl = lib.getExe' supervisor "supervisorctl";
+
+in
+{
+  microvm.virtiofsdScripts = lib.mkIf requiresVirtiofsd {
+    run =
+      let
+        supervisordConfig = pkgs.writeText "${config.networking.hostName}-virtiofsd-supervisord.conf" ''
+          [supervisord]
+          nodaemon=true
+          #user=root
```

Comment:
I think it required that explicitly but I would need to check the docs

Reply:
It's not required. That's great because that means it won't fail when the user runs it from a package instead of thru the systemd unit.